custom_components/tuya_local/devices/blitzwolf_bwsh5_humidifier.yaml

```diff
@@ -153,10 +163,24 @@ secondary_entities:
       - id: 18
         type: string
         name: option
+        optional: true
         mapping:
           - dps_val: c
             value: Celsius
             icon: "mdi:temperature-celsius"
           - dps_val: f
             value: Fahrenheit  
             icon: "mdi:temperature-fahrenheit"
+  - entity: binary_sensor
+    name: Low water
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 22
+        type: bitfield
+        name: sensor
+        mapping:
+          - dps_val: 0
```

Comment:
Leave the `dps_val` off here, so that any other value is caught by this as a default.
You should leave dp 22 on the main entity above, but rename it to fault_code, so the actual value can be read.  Probably another binary_sensor called Fault, which is false on 0 and 1, and otherwise true should be added to report other Faults (unless you know there aren't any other faults reported by the device - but usually there are at least 2 fault codes documented for dehumidifiers).

Reply:
I am not sure how this field is conveted, this is from specification:
```
{
	"abilityId": 22,
	"accessMode": "ro",
	"code": "fault",
	"description": "",
	"extensions": {
		"iconName": "icon-Panic",
		"attribute": "128"
	},
	"name": "故障告警",
	"typeSpec": {
		"label": [
			"lack_water"
		],
		"maxlen": 1,
		"type": "bitmap",
		"typeDefaultValue": 0
	}
},
```
Can "bitmap" type have any other value than 0 and 1?

Reply:
"Bitmap" can have any number of bytes. The documentation says there is only one, so adding another catch-all sensor is not needed, but in case the documentation is wrong, or a later firmware upgrade adds additional fault detection and reporting, it is still better to just catch 1 for "Tank full/missing" and map everything else (whether no bits set or bits 2 and above) to false.